blaze/compute/sql.py

```diff
@@ -842,6 +842,24 @@ def compute_up(t, s, **kwargs):
     return s.order_by(*cols)
 
 
+@dispatch(Sample, sa.Table)
+def compute_up(t, s, **kwargs):
+    n = t.n if t.n is not None else int(t.frac * s.count().scalar())
```

Comment:
This is going to actually execute a sql query at this point when you call `.scalar()`. Can we look into pushing this into a subselect to reduce the IO overhead. Also, should we special case postgres>=9.5 wich has the TABLESAMPLE clause on selects?